### DISCUSSION TITLE
String length problems

### DISCUSSION THREAD

**Discussion by ivanismr2018** [chillerlan/php-qrcode#107] [General]
First of all, **happy new year**. Second I use a translator I do not speak English.
I will be as explicit as possible, I want to render a large number of characters in the QR Code, it only allows a few and I would like to know if there is any way to render HTML code.
The essential thing is to pass at least 500 characters, it is for a file system and I want to paste it on the tower chassis.

Thanks in advance.

**`En Español`**

Primero que todo, feliz año nuevo. Segundo uso un traductor no hablo ingles.
Sere lo más explicito posible, quiero renderizar un número elevado de caractres en el Código QR, solo permite unos pocos y quisiera saber si existe algún modo de que renderice codigo HTML.
Lo escencial es pasar como mínimo 500 caracteres, es para un sistema de expedientes y quiero pegarlo en el chasi de la torre.

Gracias de Antemanos.

**Comment by codemasher** [comment]
If i'm understanding correctly you have problems storing a certain number of characters in a QR Code symbol. If you look at [this handy chart](https://www.qrcode.com/en/about/version.html), you can see which QR Code version and error correction mode is required to store the desired amount of characters. So for 500 (binary) characters you would need at least version 15 with ECC mode L. if the string contains multibyte characters, it's more, depending on the amount of mb characters, use `strlen()` (not `mb_strlen()`) to get the real string length.
On the other hand you could just use `QRCode::VERSION_AUTO` (the default) in order to circumvent any problems.
